### PR TITLE
Add single choice category selection

### LINKED ISSUE #6709: Add single category selection for Static Theme details
part of #6572 but splitting out so we (and QA) can track the change in the tag.

### PR DESCRIPTION
**Pull request by eviljeff**
Another part of #6572 - changes the category selector into radio buttons.  Fixes #6709

Most of the changes here are refactoring.

### CODE REVIEW

File: src/olympia/devhub/forms.py [hunk 1]
```diff
@@ -878,3 +879,47 @@ class DistributionChoiceForm(happyforms.Form):
...
+        choices = [
+            (c.id, c.name) for _, c in sorted_cats]
+        form.choices = choices  # sorted(choices, key=lambda x: x.name)
+
+        initials = self.addon.app_categories.get(app.id, [])
+        if len(initials) > 0:
+            form.initial = initials[0]
+
+        # If this add-on is featured for this application, category changes are
+        # forbidden.
```
Comment (diox) [inline src/olympia/devhub/forms.py]: Why? I've skimmed through the docs for the static themes and could not find a reference to this.

**Comment by diox** [comment]
Oh, could you rename the commit to indicate that it's for static themes only at the moment?

File: src/olympia/devhub/forms.py (hunk 1 above)
Comment (eviljeff) [reply src/olympia/devhub/forms.py]: Oh, that's something I pulled over from the multiple category form https://github.com/mozilla/addons-server/blob/master/src/olympia/addons/forms.py#L279 - no-one's mentioned it either way so I thought I'd keep the same behavior as extensions.

Comment (diox) [reply src/olympia/devhub/forms.py]: Ah, I had never noticed this. Not a big deal, let's keep it for now, I was just wondering.

**Pull request merged**